cirq-core/cirq/transformers/randomized_measurements.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Sequence
+from typing import Any, Literal
+
+import cirq
+import numpy as np
+import numpy.typing as npt
+from cirq.transformers import transformer_api
+
+
+@transformer_api.transformer
```

Comment:
Weren't you saying that this violates the transformer API because it returns a list of circuits instead of a single circuit? It could be easily modified to just return a single circuit (then you just call it $n$ times if you want $n$ circuits).